File: packages/sdk-snippets/src/targets/csharp/dotnet6/webhooks/fixtures/security-variables.cs
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();

// Your ReadMe secret
var secret = "my-readme-secret";

app.MapPost("/webhook", async context =>
{
  // Verify the request is legitimate and came from ReadMe.
  var signature = context.Request.Headers["readme-signature"];
  var body = await new StreamReader(context.Request.Body).ReadToEndAsync();

  try
  {
    ReadMe.Webhook.Verify(body, signature, secret);
  }
  catch (Exception e)
  {
    context.Response.StatusCode = StatusCodes.Status401Unauthorized;
    await context.Response.WriteAsJsonAsync(new
    {
      error = e.Message,
    });
    return;
  }

  // Fetch the user from the database and return their data for use with OpenAPI variables.
  // @todo Write your own query logic to fetch a user by `body["email"]`.

  await context.Response.WriteAsJsonAsync(new
  {
    // OAS Security variables
    keys = new[]
    {
      new
      {
        petstore_auth = "default-key",
      },
      new
      {
        name = "basic_auth",
        user = "user",
        pass = "pass",
      },
    }
  });
});

app.Run($"http://localhost:8000");

File: packages/sdk-snippets/src/targets/ruby/rails/webhooks/fixtures/security-variables.rb
require 'readme/webhook'

class MetricsController < ApplicationController
  def index
    render json: { message: 'hello world' }
  end

  def post
    head :ok
  end

  def webhook
    # Your ReadMe secret
    secret = 'my-readme-secret'
    # Verify the request is legitimate and came from ReadMe
    signature = request.headers['readme-signature']

    begin
      Readme::Webhook.verify(request.raw_post, signature, secret)
    rescue Readme::MissingSignatureError, Readme::ExpiredSignatureError, Readme::InvalidSignatureError => e
      # Handle invalid requests
      render json: { error: e.message }, status: 401
      return
    end

    # Fetch the user from the database and return their data for use with OpenAPI variables.
    # current_user ||= User.find(session[:user_id]) if session[:user_id]
    render json: {
      # OAS Security variables
      keys: [
        {
          petstore_auth: 'default-key',
        },
        {
          name: 'basic_auth',
          user: 'user',
          pass: 'pass',
        },
      ]
    }
  end
end
